students/swietczak_monika/lesson_07_string_datetime/date_calculator.py

```diff
@@ -0,0 +1,22 @@
+from datetime import datetime, timedelta, date
+
+today = datetime.now()
+today_date = date.today()
+
+y = int(input("How many years to add: "))
+d = int(input("How many days to add: "))
+h = int(input("How many hours to add: "))
+m = int(input("How many minutes to add: "))
+
+# there are incorrect results when d=2 or d=3
+if 1 < d < 4:
+    d += y * 365.25 + 1
+else:
+    d += y * 365.25
+new_date = today_date + timedelta(days=d)
```

Comment:
I believe your algorithm works, you can also use datetime lib to count this:
datetime.now() + timedelta(weeks=y* 52, days=d, hours=h, minutes=m)